### PR TITLE
Deselect row everytime setProperties called

### LINKED ISSUE #14830: Selected state should be cleared when data reset (Declarative Table Component)
Steps to Reproduce:

After I select the first "Warning" for the "test" db and then select the "test-2" db, the "Warning" stays selected (but I want it to be cleared and able to be selected again)

Screen Shot 2021-03-22 at 4 56 28 PM
Screen Shot 2021-03-22 at 4 56 36 PM

### PR DESCRIPTION
**Pull request by cssuh**
This PR fixes #14830

### CODE REVIEW

**Review by alanrenmsft — APPROVED** [review]

File: src/sql/workbench/browser/modelComponents/declarativeTable.component.ts [hunk 1]
```diff
@@ -245,6 +245,9 @@ export default class DeclarativeTableComponent extends ContainerBase<any, azdata
 
 	private static ACCEPTABLE_VALUES = new Set<string>(['number', 'string', 'boolean']);
 	public setProperties(properties: azdata.DeclarativeTableProperties): void {
+		if (properties.dataValues) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/modelComponents/declarativeTable.component.ts:248]: Why just dataValues - why not data as well?

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]

File: src/sql/workbench/browser/modelComponents/declarativeTable.component.ts (hunk 1 above)
Comment (cssuh) [reply src/sql/workbench/browser/modelComponents/declarativeTable.component.ts:248]: but aren't we deprecating data?

Comment (alanrenmsft) [reply src/sql/workbench/browser/modelComponents/declarativeTable.component.ts:248]: for backward compact, we should still check it
if(properties.data || properties.dataValues)

Comment (Charles-Gagnon) [reply src/sql/workbench/browser/modelComponents/declarativeTable.component.ts:248]: Yes, but that doesn't mean we shouldn't handle that now while people are still using it.

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]
And actually - you can't just check whether the properties exist. If a user updates some other property then it'll still call this and have the data/dataValues property set which means it'll unselect the row (which we don't want in that case).

You should only deselect the row if the values change, which we already have a check for down on lin 284.

**Comment by Charles-Gagnon** [comment]
Although I'm surprised that clearContainer isn't clearing the selected row too. Is there something else going on there where it keeps the selected row even when the container is cleared?

**Comment by cssuh** [comment]
yes, if you look at the issue I linked, the selection was not getting cleared.

**Review by Charles-Gagnon — APPROVED** [review]
Hmm, I'd love it if you could dig into why the row isn't being cleared when the container is cleared since that seems like a bug to me. But for fixing the issue this seems fine and shouldn't hurt anything else at least.

**Pull request merged**